force-app/main/default/classes/service/SObjectFilterQueryService.cls

```diff
@@ -0,0 +1,164 @@
+public class SObjectFilterQueryService {
+    public static List<sObject> getFilteredRecords(String sObjectName, List<FilterConditionWrapper> filterConditions, String filterLogic, Integer recordSize){
```

Comment:
Please move the method's parameters into the constructor, or use the Builder design pattern to shape them better.

Reply:
And note that the "recordSize" parameter should not be required.